src/Stratis.Bitcoin.Features.SmartContracts/Embedded/Multisig.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System.Linq;
+using NBitcoin;
+using Stratis.SmartContracts;
+using Stratis.SmartContracts.CLR;
+
+[EmbeddedContract(EmbeddedContractType.Multisig)]
+public class MultiSig : SmartContract
+{
+    const string primaryGroup = "main";
+    private readonly uint version;
+    private readonly Authentication authentication;
+
+    public MultiSig(ISmartContractState state, IPersistenceStrategy persistenceStrategy, Network network) : base(state)
+    {
+        uint version = state.Message.ContractAddress.ToUint160().GetEmbeddedVersion();
+
+        Assert(version == 1, "Only a version of 1 is supported.");
+
+        this.version = version;
+        this.authentication = new Authentication(GetState(state, persistenceStrategy, EmbeddedContractAddress.Create(typeof(Authentication), 1)), network);
```

Comment:
It seems it makes more sense to me that Authentication deployed as another smart contract and MultiSig has to make external calls to that contract.

Reply:
Authentication is being deployed as a separate contract. It's just that this seems faster and hence, why not?

Reply:
This decision bring Initialized logic and it is not what expected in regular design for smart contracts.

Reply:
The Initialized flag is required due to the absence of a deployment step for embedded smart contracts. Also, embedded contracts obtain access to all full node services via non-default constructors. These things are different for embedded contracts and should not to be expected to match the regular design for smart contracts.